File: android/src/main/kotlin/com/mytiki/sdk/capture/receipt/capacitor/rsp/RspRetailerAccount.kt
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */

package com.mytiki.sdk.capture.receipt.capacitor.rsp

import com.microblink.linking.Account
import com.mytiki.sdk.capture.receipt.capacitor.RetailerEnum
import org.json.JSONObject

class RspRetailerAccount(
    val account: Account,
    val isVerified: Boolean
) : Rsp {
    override fun toJson(): JSONObject =
        JSONObject()
            .put("username", account.credentials.username())
            .put("retailer", RetailerEnum.fromValue(account.retailerId).toString())
            .put("isVerified", isVerified)
}

File: ios/Plugin/Rsp/RspRetailerAccount.swift
//
/*
 * Copyright (c) TIKI Inc.
 * MIT license. See LICENSE file in root directory.
 */

import Foundation
import BlinkReceipt
import BlinkEReceipt
import Capacitor


struct RspRetailerAccount : Rsp{
    var account: RspAccount
    var isVerified: Bool
    
    func toJson() -> JSObject {
        JSObject.updateValue("username", account.credentials.username())
        JSObject.updateValue("retailer", RetailerEnum.fromInt(account.retailerId).toString())
        JSObject.updateValue("isVerified", isVerified)
    }
}
